### PR TITLE
DSEGOG-129 save sessions

### PR DESCRIPTION
**Pull request by joshdimanteto**
## Description
- A user should be able to create a user session filling out a dialog form that is opened by clicking the add Icon located in the drawer.
- The save and save as buttons are added as placeholder in this PR function will be added into a later PR.

## Testing instructions
- [x] Check that the request body send to the backend matches the current state of the website
- [x] Check the error handling in the dialog

- [x] Review code
- [x] Check Actions build
- [x] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-129

### CODE REVIEW

**Comment by joshdimanteto** [comment]
save-session-1

**Comment by joshdimanteto** [comment]
add save session

File: src/session/saveSessionDialogue.component.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      setErrorMessage('Please enter a name');
+    }
+  }, [handleClose, saveSession, sessionName, sessionSummary, state]);
+
+  return (
+    <Dialog open={open} onClose={onClose} maxWidth="lg">
+      <DialogTitle>Save Session</DialogTitle>
+      <DialogContent>
+        <TextField
+          label="Name*"
```
Comment (louise-davies) [inline src/session/saveSessionDialogue.component.tsx]: Minor point - but the `*` shouldn't be added to the `label` - instead add the `required` prop to the `TextField` - this adds the `*` automatically without then messing up accessibility (i.e. the * is marked with aria-hidden so screen readers ignore it, etc.)

**Review by louise-davies — CHANGES_REQUESTED** [review]
Looks fine other than one comment I had re: a text field.

**Review by louise-davies — APPROVED** [review]
LGTM

**Pull request merged**